addons/binding/org.openhab.binding.velux/src/main/java/org/openhab/binding/velux/bridge/comm/BCidentifyProduct.java

```diff
@@ -0,0 +1,167 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.velux.bridge.comm;
+
+import org.openhab.binding.velux.bridge.VeluxBridge;
+
+/**
+ * I/O messages supported by the <B>Velux</B> bridge.
+ * It provides informations how to send query through the
+ * {@link org.openhab.binding.velux.handler.VeluxBridgeHandler VeluxBridgeHandler}.
+ *
+ * @author Guenther Schreiner - Initial contribution.
+ */
+
+public class BCidentifyProduct implements BridgeCommunicationProtocol<BCidentifyProduct.Response> {
```

Comment:
It seems like almost all `BridgeCommunicationProtocol ` classes are basically identical. Only the type of response data, the action name and some request parameters will change.
I suggest to simplify all these classes and provide only one implementation of `Request`, `Response` and parameterise the actual request by the action name, the parameter object and a type information for the response data. This way you will save a huge amount of duplicate code.

Reply:
Have reduced the similar code fragments (now for JSON as well as for SLIP protocol).